pkg/channel/message_dispatcher.go

```diff
@@ -190,7 +191,7 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context, url *url.URL
 	}
 
 	if span.IsRecordingEvents() {
-		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders, kncloudevents.PopulateSpan(span))
+		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders, tracing.PopulateSpan(span, url.String()), kncloudevents.PopulateSpan(span))
```

Comment:
Do we really need to have both `PopulateSpan` function? If we need both attributes, could you just call one from the other, so that there is a single entry for tracing? WDYT @slinkydeveloper

Reply:
+1 for this one, no need to have 2 transformers for that

Reply:
Yep, that was my initial approach but ultimately decided against it for these reasons;
* would add a dependency between `kncloudevents` & `tracing`
* with the separate approach, they can both leverage the same method signature quite easily

However, happy to reconsider if you have an approach that you feel would be clearer :)

Reply:
I think unless there are cases where only one should be used, we should keep them in the same function, especially considering the fact that they are in spec.

Reply:
> would add a dependency between kncloudevents & tracing

That's not really a problem, data plane code that uses kncloudevents also uses tracing to implement the tracing, so...

> with the separate approach, they can both leverage the same method signature quite easily

TBH I don't see a use case where users of that code would need to use only one of them